### PR TITLE
Add support for importing TRN via user import

### PR DESCRIPTION
**Pull request by hortha**
### Context

We need to be able to import TRN as part of the user import

### Changes proposed in this pull request

Add ability to insert user with TRN or update user with missing TRN via user import

### Guidance to review

Database changes, backend processing, page tweaks + tests.

### Checklist

-   [x] Attach to Trello card
-   [x] Rebased master
-   [x] Cleaned commit history
-   [x] Tested by running locally
-   [x] Reminder created to manually clean any removed app settings post deployment

### CODE REVIEW

**Comment by hortha** [comment]
[Identity user import - include TRN](https://trello.com/c/xeqZjYqw/682-identity-user-import-include-trn)

File: dotnet-authserver/src/TeacherIdentity.AuthServer/Pages/Admin/UserImport.cshtml.cs [hunk 1]
```diff
@@ -84,7 +96,8 @@ public async Task<IActionResult> OnGetDownloadFile()
                     RowNumber = r.RowNumber,
                     Id = r.Id,
                     UserId = r.UserId,
-                    Errors = r.Errors is null ? null : string.Join(".", r.Errors),
+                    UserImportRowResult = r.UserImportRowResult,
+                    Notes = r.Notes is null ? null : string.Join(".", r.Notes),
```
Comment (gunndabad) [inline dotnet-authserver/src/TeacherIdentity.AuthServer/Pages/Admin/UserImport.cshtml.cs:100]: Just noticed this...
```suggestion
                    Notes = r.Notes is null ? null : string.Join(". ", r.Notes),
```

File: dotnet-authserver/src/TeacherIdentity.AuthServer/Services/UserImport/UserImportProcessor.cs [hunk 2]
```diff
@@ -127,6 +128,14 @@ public async Task Process(Guid userImportJobId)
                 {
                     errors.Add($"{UserImportRow.DateOfBirthHeader} field should be a valid date in ddMMyyyy format");
                 }
+
+                if (!string.IsNullOrEmpty(row.Trn))
+                {
+                    if (!Regex.IsMatch(row.Trn, @"^\d{7}$"))
```
Comment (gunndabad) [inline dotnet-authserver/src/TeacherIdentity.AuthServer/Services/UserImport/UserImportProcessor.cs:134]:
```suggestion
                if (!string.IsNullOrEmpty(row.Trn) && !Regex.IsMatch(row.Trn, @"^\d{7}$"))
                {
```

File: dotnet-authserver/src/TeacherIdentity.AuthServer/Services/UserImport/UserImportProcessor.cs [hunk 3]
```diff
@@ -200,11 +219,66 @@ public async Task Process(Guid userImportJobId)
...
-                    errors.Add("A user already exists with the specified email address");
 
                     User existingUser = await _dbContext.Users.SingleAsync(u => u.EmailAddress == row!.EmailAddress);
-                    userImportJobRow.UserId = existingUser.UserId;
-                    userImportJobRow.Errors = errors;
+                    if (!string.IsNullOrEmpty(row!.Trn))
+                    {
+                        if (existingUser.Trn is null)
+                        {
+                            existingUser.Trn = row.Trn;
```
Comment (gunndabad) [inline dotnet-authserver/src/TeacherIdentity.AuthServer/Services/UserImport/UserImportProcessor.cs:228]: Need to issue a `UserUpdatedEvent` if we're changing a user

**Review by gunndabad — APPROVED** [review]

File: dotnet-authserver/src/TeacherIdentity.AuthServer/Services/UserImport/UserImportProcessor.cs (hunk 2 above)
Comment (hortha) [reply dotnet-authserver/src/TeacherIdentity.AuthServer/Services/UserImport/UserImportProcessor.cs]: Done

File: dotnet-authserver/src/TeacherIdentity.AuthServer/Pages/Admin/UserImport.cshtml.cs (hunk 1 above)
Comment (hortha) [reply dotnet-authserver/src/TeacherIdentity.AuthServer/Pages/Admin/UserImport.cshtml.cs]: Done

**Pull request merged**